views/js/runner/providerRegistry.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2014 (original work) Open Assessment Technlogies SA (under the project TAO-PRODUCT);
+ *
+ */
+/**
+ * @author Sam <sam@taotesting.com>
+ */
+define(['lodash'], function(_){
+    'use strict';
+    
+    /**
+     * Transfor the target into a a provider registry
+     * It adds two methods registerProvider() and getProvider();
+     * 
+     * @todo useful to move it to tao/core ?
+     * @param {Object} target
+     * @param {Function} [validator] - a function to validate the provider to be registered
+     * @returns {target} the target itself
+     */
+    function providerRegistry(target, validator){
```

Comment:
This code has been adapted from the item runner. Maybe it would make sense to move it to tao-core if this becomes a pattern we encourage using ?